hbase-server/src/main/java/org/apache/hadoop/hbase/master/RegionsRecoveryChore.java

```diff
@@ -171,4 +167,20 @@ private void prepareTableToReopenRegionsMap(
 
   }
 
+  // hashcode/equals implementation to ensure at-most one object of RegionsRecoveryChore
```

Comment:
This seems very subtle but in general I think there should be an invariant somewhere that guarantees this rather than silently relying on hashCode() implementations. I'll let the committer take a deeper look at this since I'm new here.

Reply:
Ya I got that. What I meant was that the behavior is very subtle. Tomorrow if someone wants to have multiple instances of this chore and try submitting them, they get overwritten. Ideally one should get an error saying that only one instance of it should or running or something like that.

Reply:
Sounds good but ideally no chore should have multiple instances running anyways.